runtime/src/bank.rs

```diff
@@ -556,6 +557,17 @@ impl Bank {
         self.store_account(pubkey, &new_account);
     }
 
+    fn update_incinerator(&self) {
+        let account = self.get_sysvar_account(&sysvar::incinerator::id());
+        if let Some(account) = account {
+            if account.lamports > 0 {
+                self.capitalization
+                    .fetch_sub(account.lamports, Ordering::Relaxed);
+            }
+            self.store_account(&sysvar::incinerator::id(), &Account::default());
```

Comment:
Maybe, this will reset `account.owner`. Better off using `sysvar.create_account()` machinery.

Also, Account::default() is `lamport == 0`, so, the account can vanish at any moment, even if we set owner correctly.

Sorry, so much pitfalls are laying around AccountsDB.. ;)

Reply:
Hmm, but I want the incinerator account to vanish so `Account::default()`/`lamport  == 0` is exactly right IMO

Reply:
I see! Thanks for explaining.

Still, I think there is still small problem with `Account::default()`. That's because `Sysvar1nc1nerator11111111111111111111111111` isn't `rent-exempt` with `Account::default()`: https://github.com/solana-labs/solana/blob/de9d8cd849040e556ae52c6b77902aa4e563b4be/runtime/src/rent_collector.rs#L39

So, the problem is that there is very corner case of not being able to exactly burn the specified amount of lamports. Say, someone wanted to burn 1234 lamports but `Sysvar1nc1nerator11111111111111111111111111` isn't rent-exempt so `1` lamport is collected when paying to it. So the burnt capitization will be `1233`. and the `1` lamport is now part of stake-weighted reward pool for a slot.

I guess we just need to adjust that quoted condition in the RentCollector. :)

Also as for `account.owner`, I thought all accounts starting `sysvar.*` must be owner for consistency. Or, we just call `1nc1nerator11111111111111111111111111`?

Reply:
> Also as for `account.owner`, I thought all accounts starting `sysvar.*` must be owner for consistency. Or, we just call `1nc1nerator11111111111111111111111111`?

I don't mind just calling this `1nc1nerator11111111111111111111111111`.   You're right, it doesn't behave quite like a sysvar.

Reply:
Also, I noticed `solana_sdk::sysvar::is_sysvar_id(pubkey))` and `sysvar::check_id(&account.owner)` are similar but these are different implementations... Dunno how these usages are differentiated...